lib/maths/CBoostedTreeImpl.cc

```diff
@@ -725,11 +737,18 @@ bool CBoostedTreeImpl::selectNextHyperparameters(const TMeanVarAccumulator& loss
     if (m_FeatureBagFractionOverride == boost::none) {
         parameters(i++) = m_FeatureBagFraction;
     }
+    LOG_TRACE(<< "loss moments = " << lossMoments << " for lambda = " << m_Lambda
+              << ", gamma = " << m_Gamma << ", eta = " << m_Eta
+              << ", eta growth rate per tree = " << m_EtaGrowthRatePerTree
+              << ", feature bag fraction = " << m_FeatureBagFraction);
 
     double meanLoss{CBasicStatistics::mean(lossMoments)};
-    double lossVariance{CBasicStatistics::variance(lossMoments)};
+    double meanLossVariance{CBasicStatistics::variance(lossMoments) /
+                            CBasicStatistics::count(lossMoments)};
+    m_MeanLossVariance.add(meanLossVariance);
 
-    bopt.add(parameters, meanLoss, lossVariance);
+    bopt.add(parameters, meanLoss + std::sqrt(meanLossVariance),
+             CBasicStatistics::mean(m_MeanLossVariance));
```

Comment:
Ok I admit this is a bit hacky. This is what I was referring to in this points:
> By using the local estimate of variance we could get stuck exploring regions where the error is relatively low and the variance across the folds is relatively high since the mass GP has significant spread here and so there is a chance assigned for large improvements. This is undesirable because we typically want to avoid regions where we can't estimate the validation error well. Item 2 mitigates this somewhat, but I have also switched to just using the mean error variance overall.

Basically, if we say that the surface is uncertain at some point there is a significant chance of low values in the vicinity of that point, in the sense that the GP has non-zero mass for low values. This means the EI search can get fixated testing values near this point. Chances are these also have significant error. But we really don't want it to do this, since we don't know if these are good or bad points. I wanted some way of capturing the error in the surface values, which didn't display this problem. This was the best I came up with. Happy if you can think of better alternatives also to discuss further offline.